src/prometheus_time.erl

```diff
@@ -155,14 +155,13 @@ maybe_convert_to_native(DU, Value) ->
         _ -> to_native(Value, DU)
     end.
 
--spec maybe_convert_to_du(duration_unit(), infinity | number()) -> infinity | number().
+-spec maybe_convert_to_du(undefined | duration_unit(), infinity | number()) -> infinity | number().
+maybe_convert_to_du(undefined, Value) ->
+    Value;
 maybe_convert_to_du(_, infinity) ->
     infinity;
```

Comment:
Did (`undefined`, `infinity`) ever make sense before? Because we're losing that with this change.

Reply:
Very good question. Just checking, coverage had actually never hit this clause. This is called from gauges, histograms, summaries, and quantile_summaries. Quantile summaries is the work in this PR, and it cannot be `infinity`. Summaries are doing number addition with `+` before giving in to this helper, so it cannot be an atom either. Histograms are doing `+` or `lists:sum/1`, so again must be integers. Interestingly, only gauges can actually pass an atom as the second argument, and they'll pass `undefined`, which would crash, but that wasn't tested. Adding a test for that.

Also, `(undefined, infinity)` would match the second clause, which just as before, would return infinity, so no real functional change is lost.

Reply:
> (undefined, infinity) would match the second clause

You mean in the past? Because as is (check above) it wouldn't, right? As soon as it's `undefined` you get clause `(undefined, Value)`.

In any case, I'm Ok with the change and since you're gonna put some tests on, even better.

Reply:
I mean, as soon as the first argument is undefined, it'll match the first clause, and return `Value`, which, if the function was called with `(undefined, infinity)`, it would return `infinity`, which is the same effect that `(undefined, infinity)` was having before, so no change of behaviour there 😄 

In any case, see the new tests in the recent commits :)

Reply:
True. Good stuff.